ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestinationFactory.java

```diff
@@ -27,21 +28,25 @@
 public class AzureCloudDestinationFactory implements CloudDestinationFactory {
 
   private static final Logger logger = LoggerFactory.getLogger(AzureCloudDestinationFactory.class);
+  private final CloudConfig cloudConfig;
   private final AzureCloudConfig azureCloudConfig;
   private final AzureMetrics azureMetrics;
 
   public AzureCloudDestinationFactory(VerifiableProperties verifiableProperties, MetricRegistry metricRegistry) {
+    this.cloudConfig = new CloudConfig(verifiableProperties);
     this.azureCloudConfig = new AzureCloudConfig(verifiableProperties);
     azureMetrics = new AzureMetrics(metricRegistry);
   }
 
   @Override
   public CloudDestination getCloudDestination() throws IllegalStateException {
     try {
-      return new AzureCloudDestination(azureCloudConfig, azureMetrics);
+      AzureCloudDestination dest = new AzureCloudDestination(cloudConfig, azureCloudConfig, azureMetrics);
+      dest.testAzureConnectivity();
+      return dest;
     } catch (Exception e) {
       logger.error("Initializing Azure destination", e.getMessage());
```

Comment:
minor, change this to something like `logger.error("Error occurred when initializing Azure destination : {}", e.getMessage());`

Reply:
ok